pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -695,8 +695,10 @@ type RolloutAnalysisBackground struct {
 
 // RolloutAnalysis defines a template that is used to create a analysisRun
 type RolloutAnalysis struct {
-	//Templates reference to a list of analysis templates to combine for an AnalysisRun
-	Templates []RolloutAnalysisTemplate `json:"templates,omitempty" protobuf:"bytes,1,rep,name=templates"`
+	// Templates reference to a list of analysis templates to combine for an AnalysisRun
+	// +patchMergeKey=templateName
+	// +patchStrategy=merge
+	Templates []RolloutAnalysisTemplate `json:"templates,omitempty" patchStrategy:"merge" patchMergeKey:"templateName" protobuf:"bytes,1,rep,name=templates"`
```

Comment:
PR title and commit message says merge key is added to Analysis args but the code is adding it in the AnalysisTemplate.
Which one is correct? Just want to make sure this is added at the right place.

Reply:
Thanks for the catch, yes it is templates